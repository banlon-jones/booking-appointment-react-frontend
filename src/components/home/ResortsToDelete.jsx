import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import axios from 'axios';
import classes from './Resorts.module.css';

import { resortsFetched } from '../../store/resorts/resortsSlice';
import CircleSpinner from '../spinners/CircleSpinner';

const ResortsToDelete = () => {
  const resorts = useSelector((state) => state.resorts);
  const dispatch = useDispatch();

  const [errorMessage, setErrorMessage] = useState('');
  const [loading, setLoading] = useState(false);

  const fetchResorts = async () => {
    try {
      setLoading(true);
      const { data } = await axios({
        url: 'https://resorts-booking-api.herokuapp.com/resorts',
        method: 'GET',
        headers: {
          authorization: sessionStorage.getItem('JwtAccessToken'),
        },
      });
      if (data.error) {
        setErrorMessage(data.error);
        return;
      }
      dispatch(resortsFetched(data));
    } catch (err) {
      setLoading(false);
    }
  };

  const deleteResort = async (id) => {
    console.log('resort id is: ', id);
    await axios({
      url: `https://resorts-booking-api.herokuapp.com/resorts/${id}`,
      method: 'DELETE',
      headers: {
        authorization: sessionStorage.getItem('JwtAccessToken'),
      },
    });
    fetchResorts();
  };

  useEffect(() => {
    if (resorts.length === 0) fetchResorts();
  }, [resorts]);

  return (
    <div>
      <ul className={classes.resorts_to_delete_container}>
        <h2 className="text-center w-100">All Resorts</h2>
        {resorts.map((resort) => (
          <li key={resort.id} className={classes.resort_to_delete}>
            {resort.name}
            <button
              type="button"
              onClick={() => deleteResort(resort.id)}
              className={classes.btn_delete_resort}
            >
              delete
            </button>
          </li>
        ))}
      </ul>
      {errorMessage && <h2>{errorMessage}</h2>}
      {loading && <CircleSpinner />}
    </div>
  );
};

export default ResortsToDelete;
