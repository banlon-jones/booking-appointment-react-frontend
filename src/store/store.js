/* eslint-disable import/no-named-as-default */
/* eslint-disable import/no-named-as-default-member */
import { configureStore } from '@reduxjs/toolkit';
import userSlice from './user/user';
import resortDetailSlice from './resort-details/resortDetail';
import newResortSlice from './resort-details/newResort';
import updateResortSlice from './resort-details/updateResort';
import resortsSlice from './resorts/resortsSlice';
import reservationSlice from './reservation/reservation';

const store = configureStore({
  reducer: {
    user: userSlice,
    resortDetail: resortDetailSlice,
    newResort: newResortSlice,
    updateResort: updateResortSlice,
    resorts: resortsSlice,
    reservations: reservationSlice,
  },
});

export default store;
