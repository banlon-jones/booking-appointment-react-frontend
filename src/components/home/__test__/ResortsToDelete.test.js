import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';

import store from '../../../store/store';
import ResortsToDelete from '../ResortsToDelete';

describe('ResortsToDelete snapshots', () => {
  const initialState = { resorts: [] };
  const mockStore = configureStore();
  let store;

  it('matches ResortsToDelete snapshot', () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <ResortsToDelete />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Page rendering', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <ResortsToDelete />
        </Router>
      </Provider>,
    );
  });

  test('renders the ResortsToDelete page heading', () => {
    expect(screen.getByRole('heading')).toHaveTextContent(/All Resorts/);
  });

  test('renders  Loading spinner in the ResortsToDelete page', () => {
    expect(screen.getAllByText('Loading...'));
  });
});
