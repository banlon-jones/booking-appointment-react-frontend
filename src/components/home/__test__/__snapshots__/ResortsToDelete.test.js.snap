// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResortsToDelete snapshots matches ResortsToDelete snapshot 1`] = `
<div>
  <ul
    className="resorts_to_delete_container"
  >
    <h2
      className="text-center w-100"
    >
      All Resorts
    </h2>
  </ul>
</div>
`;
